lib/gui/components/update-notifier.js

```diff
@@ -138,17 +166,48 @@ module.exports = function($http, $q, ModalService, UPDATE_NOTIFIER_SLEEP_DAYS, M
    */
   this.notify = () => {
     return this.getLatestVersion().then((version) => {
-      return ModalService.open({
-        template: './components/update-notifier/templates/update-notifier-modal.tpl.html',
-        controller: 'UpdateNotifierController as modal',
-        size: 'update-notifier',
-        resolve: {
-          options: _.constant({
-            version
-          })
+      settings.set('lastUpdateNotify', Date.now());
+      settings.set('sleepUpdateCheck', false);
```

Comment:
Looking at it `lastUpdateNotify` is set before anytime the dialog appears so it can't be relied upon alone as-is, because anytime the dialog appears it'd pause it for seven days without the help of `sleepUpdateCheck`. However it doesn't need to be set at the start of the dialog code, we can instead set it if `results.sleepUpdateCheck` is `true`. We don't actually use `lastUpdateNotify` for anything else so this should work fine, unless there are plans to do something like log the space between each update check. I believe this should work:

```diff
diff --git a/lib/gui/components/update-notifier.js b/lib/gui/components/update-notifier.js
index 9fad229..92dfbf0 100644
--- a/lib/gui/components/update-notifier.js
+++ b/lib/gui/components/update-notifier.js
@@ -142,12 +142,11 @@ UpdateNotifier.service('UpdateNotifierService', function(
   this.shouldCheckForUpdates = () => {
     const lastUpdateNotify = settings.get('lastUpdateNotify');
 
-    if (!settings.get('sleepUpdateCheck') || !lastUpdateNotify) {
+    if (!lastUpdateNotify) {
       return true;
     }
 
     if (lastUpdateNotify - Date.now() > units.daysToMilliseconds(this.UPDATE_NOTIFIER_SLEEP_DAYS)) {
-      settings.set('sleepUpdateCheck', false);
       return true;
     }
 
@@ -166,8 +165,6 @@ UpdateNotifier.service('UpdateNotifierService', function(
    */
   this.notify = () => {
     return this.getLatestVersion().then((version) => {
-      settings.set('lastUpdateNotify', Date.now());
-      settings.set('sleepUpdateCheck', false);
 
       return new Bluebird((resolve) => {
         const BUTTONS = [
@@ -194,7 +191,9 @@ UpdateNotifier.service('UpdateNotifierService', function(
           });
         });
       }).then((results) => {
-        settings.set('sleepUpdateCheck', results.sleepUpdateCheck);
+        if (results.sleepUpdateCheck) {
+          settings.set('lastUpdateNotify', Date.now());
+        }
 
         AnalyticsService.logEvent('Close update modal', {
           sleepUpdateCheck: results.sleepUpdateCheck,
```